django/db/backends/sqlite3/_functions.py

```diff
@@ -0,0 +1,471 @@
+    dt = _sqlite_datetime_parse(dt, tzname, conn_tzname)
+    if dt is None:
+        return None
+    if lookup_type == 'week_day':
+        return (dt.isoweekday() % 7) + 1
+    elif lookup_type == 'iso_week_day':
+        return dt.isoweekday()
+    elif lookup_type == 'week':
+        return dt.isocalendar()[1]
+    elif lookup_type == 'quarter':
+        return ceil(dt.month / 3)
+    elif lookup_type == 'iso_year':
+        return dt.isocalendar()[0]
+    else:
+        return getattr(dt, lookup_type)
+
+
+def _sqlite_datetime_trunc(lookup_type, dt, tzname, conn_tzname):
+    dt = _sqlite_datetime_parse(dt, tzname, conn_tzname)
+    if dt is None:
+        return None
+    if lookup_type == 'year':
+        return "%i-01-01 00:00:00" % dt.year
+    elif lookup_type == 'quarter':
+        month_in_quarter = dt.month - (dt.month - 1) % 3
+        return '%i-%02i-01 00:00:00' % (dt.year, month_in_quarter)
+    elif lookup_type == 'month':
+        return "%i-%02i-01 00:00:00" % (dt.year, dt.month)
+    elif lookup_type == 'week':
+        dt = dt - timedelta(days=dt.weekday())
+        return "%i-%02i-%02i 00:00:00" % (dt.year, dt.month, dt.day)
+    elif lookup_type == 'day':
+        return "%i-%02i-%02i 00:00:00" % (dt.year, dt.month, dt.day)
+    elif lookup_type == 'hour':
+        return "%i-%02i-%02i %02i:00:00" % (dt.year, dt.month, dt.day, dt.hour)
+    elif lookup_type == 'minute':
+        return "%i-%02i-%02i %02i:%02i:00" % (dt.year, dt.month, dt.day, dt.hour, dt.minute)
+    elif lookup_type == 'second':
+        return "%i-%02i-%02i %02i:%02i:%02i" % (dt.year, dt.month, dt.day, dt.hour, dt.minute, dt.second)
+
+
+def _sqlite_time_extract(lookup_type, dt):
+    if dt is None:
+        return None
+    try:
+        dt = typecast_time(dt)
+    except (ValueError, TypeError):
+        return None
+    return getattr(dt, lookup_type)
+
+
+def _sqlite_prepare_dtdelta_param(conn, param):
+    if conn in ['+', '-']:
+        if isinstance(param, int):
+            return timedelta(0, 0, param)
+        else:
+            return typecast_timestamp(param)
+    return param
+
+
+def _sqlite_format_dtdelta(conn, lhs, rhs):
+    """
+    LHS and RHS can be either:
+    - An integer number of microseconds
+    - A string representing a datetime
+    - A scalar value, e.g. float
+    """
+    if lhs is None or rhs is None:
+        return None
+    conn = conn.strip()
+    try:
+        real_lhs = _sqlite_prepare_dtdelta_param(conn, lhs)
+        real_rhs = _sqlite_prepare_dtdelta_param(conn, rhs)
+    except (ValueError, TypeError):
+        return None
+    if conn == '+':
+        # typecast_timestamp returns a date or a datetime without timezone.
+        # It will be formatted as "%Y-%m-%d" or "%Y-%m-%d %H:%M:%S[.%f]"
+        out = str(real_lhs + real_rhs)
+    elif conn == '-':
+        out = str(real_lhs - real_rhs)
+    elif conn == '*':
+        out = real_lhs * real_rhs
+    else:
+        out = real_lhs / real_rhs
+    return out
```

Comment:
> Avoid unnecessary elifs.

I don't believe this is an improvement. It will give the same byte code and it is less clear to the reader that these are mutually exclusive.

> Raise NotImplementedError in the default case for unknown operators.

It's a pretty convoluted chain of events to get to a case where the operator doesn't exist - the user has to create a custom `Func` or SQL that calls this function, then they pass in bad arguments. But sure I guess.